tests/foreman/api/test_puppetmodule.py

```diff
@@ -0,0 +1,114 @@
+"""Unit tests for the ``katello/api/v2/puppet_modules`` paths."""
+from nailgun import entities
+from robottelo.common.constants import PUPPET_MODULE_NTP_PUPPETLABS
+from robottelo.common.decorators import skip_if_bug_open
+from robottelo.common.helpers import get_data_file
+from robottelo.test import APITestCase
+
+
+class RepositorySearchTestCase(APITestCase):
+    """Tests that search for puppet modules and filter by repository."""
+
+    @classmethod
+    def setUpClass(cls):
+        """Create a product. Make it available as ``cls.product``."""
+        cls.product = entities.Product().create()
+
+    def setUp(self):
+        """Create a puppet repository as ``self.repository``.
+
+        The repository belongs to ``cls.product``.
+
+        """
+        self.repository = entities.Repository(
+            content_type='puppet',
+            product=self.product,
+        ).create()
+
+    @skip_if_bug_open('bugzilla', 1260206)
+    def test_0(self):
```

Comment:
Done. I've renamed the methods like so:

* `test_0` → `test_0_search_results`
* `test_1` → `test_1_search_result`